src/smolagents/agents.py

```diff
@@ -251,7 +260,11 @@ def __init__(
 
         self.max_steps = max_steps
         self.step_number = 0
-        self.grammar = grammar
+        if grammar is not None:
+            warnings.warn(
+                "The `grammar` argument is deprecated and will have no effect on agent behavior.",
```

Comment:
Better specifying the removal version.
```suggestion
                "Argument 'grammar' is deprecated and will be removed in version 1.20. ",
```

Reply:
Done.